File: system/gd/rust/shim/src/init_flags.rs
#[cxx::bridge(namespace = bluetooth::common::init_flags)]
#[allow(unsafe_op_in_unsafe_fn)]
mod ffi {
    struct InitFlagWithValue {
        flag: &'static str,
        value: String,
    }

    extern "Rust" {
        fn load(flags: Vec<String>);
        fn set_all_for_testing();

        fn dump() -> Vec<InitFlagWithValue>;

        fn get_hci_adapter() -> i32;
        fn use_unified_connection_manager_is_enabled() -> bool;
    }
}

use crate::init_flags::ffi::InitFlagWithValue;

fn dump() -> Vec<InitFlagWithValue> {
    bt_common::init_flags::dump()
        .into_iter()
        .map(|(flag, value)| InitFlagWithValue { flag, value })
        .collect()
}

use bt_common::init_flags::*;

File: system/test/common/init_flags.cc


#include "common/init_flags.h"

#include <map>
#include <string>

namespace bluetooth {
namespace common {

bool InitFlags::logging_debug_enabled_for_all = false;
std::unordered_map<std::string, bool> InitFlags::logging_debug_explicit_tag_settings = {};
void InitFlags::Load(const char** flags) {}
void InitFlags::SetAll(bool value) { InitFlags::logging_debug_enabled_for_all = value; }

}  // namespace common
}  // namespace bluetooth
